common/lib/xmodule/xmodule/modulestore/tests/test_publish.py

```diff
@@ -139,3 +152,614 @@ def test_publish_draft_delete(self):
+
+    def assertOLXMissing(self, block_type, block_id, **kwargs):
+        """
+        Check that the course has been exported. If not, export it, then call the check.
+        """
+        self._export_if_not_already()
+        super(UniversalTestProcedure, self).assertOLXMissing(block_type, block_id, **kwargs)
+
+    def _assertOLXBase(self, block_list, draft, **kwargs):
+        """
+        Check that all blocks in the list are draft blocks in the OLX format when the course is exported.
+        """
+        for block_data in block_list:
+            block_params = self.course_db.get(block_data)
+            self.assertIsNotNone(block_params)
+            (block_type, block_id) = block_data
+            regex_func = None
+            if block_type == 'chapter':
+                regex_func = self._make_chapter_matching_regex
+            elif block_type == 'sequential':
+                regex_func = self._make_sequential_matching_regex
+                block_params.update({'seq_children_published': kwargs.get('seq_children_published')})
+            elif block_type == 'vertical':
+                regex_func = self._make_vertical_matching_regex
+            elif block_type == 'html':
+                regex_func = self._make_html_unit_matching_regex
+            block_regex = regex_func(
+                self.course_key,
+                draft=draft,
+                **block_params
+            )
+            self.assertOLXContent(block_type, block_id, draft=draft, xml_re=block_regex)
+            self.assertOLXMissing(block_type, block_id, draft=(not draft))
+
+    def assertOLXIsDraft(self, block_list, **kwargs):
+        """
+        Check that all blocks in the list are draft blocks in the OLX format when the course is exported.
+        """
+        self._assertOLXBase(block_list, draft=True, **kwargs)
+
+    def assertOLXIsPublished(self, block_list, **kwargs):
+        """
+        Check that all blocks in the list are published blocks in the OLX format when the course is exported.
+        """
+        self._assertOLXBase(block_list, draft=False, **kwargs)
+
+    def publish(self, block_type, block_id):
+        """
+        Get an item, publish it, and shift to a new course export dir.
+        """
+        # Get the specified test item from the draft branch.
+        with self.store.branch_setting(ModuleStoreEnum.Branch.draft_preferred):
+            test_item = self.store.get_item(self.course_key.make_usage_key(block_type=block_type, block_id=block_id))
+        # Publish the draft item to the published branch.
+        self.store.publish(test_item.location, self.user_id)
+        # Since the elemental operation is now complete, shift to the post-operation export directory name.
+        self.export_dir = self.EXPORTED_COURSE_AFTER_DIR_NAME
+        self.before_publish = False
+
+
+@ddt.ddt
+class ElementalPublishingTests(UniversalTestProcedure):
```

Comment:
@cpennington Care to give this class a look? Is this amount of detail in the tests clear enough? I'm at a point where I consider these tests for `publish()` are done and I'm moving on to the other operations. Any comments?